### PR TITLE
Patch artifact bucket

### PR DESCRIPTION
**Pull request by DanielSanchez97**
Fix for the artifact bucket issue

### CODE REVIEW

File: src/cdev/commands/initializer.py [hunk 1]
```diff
@@ -58,14 +61,26 @@ def create_project_cli(args) -> None:
         template_name = None
```
Comment (lo9i) [inline src/cdev/commands/initializer.py:61]: You can make this code more readable by simply assigning the template_name = args.template
And checking if it is not None and it is in the list of valid templates

File: src/cdev/commands/initializer.py [hunk 2]
```diff
@@ -149,56 +177,123 @@ def create_project(project_name: str, base_directory: DirectoryPath = None) -> N
...
+
+    Returns:
+        Dict[str, str]: Settings completed by the user
+    """
+    rv = {}
+
+    _artifact_bucket = Prompt.ask(prompt="Name of bucket to store artifacts")
+
+    if not _artifact_bucket:
+        print(f"Must provide a bucket for artifacts")
```
Comment (lo9i) [inline src/cdev/commands/initializer.py:191]: It seems that we have to do something if the _artifacts_bucket is None?

File: src/cdev/commands/initializer.py [hunk 3]
```diff
@@ -149,56 +177,123 @@ def create_project(project_name: str, base_directory: DirectoryPath = None) -> N
...
+    a base settings file, and applies any base settings to the generate file.
 
-    with open(os.path.join(base_settings_folder, f"base_settings.py"), "w"):
-        pass
+    Args:
+        file_path (FilePath): path of the file
+        base_settings (Dict[str, str], optional): Settings for base environment. Defaults to None.
+    """
+    # The settings are dynamically importable python modules, so the folder needs to be a python module
+    _touch_file(os.path.join(os.path.dirname(file_path), f"__init__.py"))
```
Comment (lo9i) [inline src/cdev/commands/initializer.py:238]: You don't need the 'f'

File: src/cdev/commands/initializer.py [hunk 4]
```diff
@@ -149,56 +177,123 @@ def create_project(project_name: str, base_directory: DirectoryPath = None) -> N
...
-        with open(
-            os.path.join(base_settings_folder, f"{environment}_settings.py"), "w"
-        ):
-            pass
+    if base_settings:
+        _render_settings_file(file_path, base_settings)
 
-        os.mkdir(os.path.join(base_settings_folder, f"{environment}_secrets"))
 
+def _render_settings_file(file_path: FilePath, base_settings: Dict[str, str]):
```
Comment (lo9i) [inline src/cdev/commands/initializer.py:246]: Also missing the return type hint

File: src/cdev/commands/initializer.py [hunk 5]
```diff
@@ -149,56 +177,123 @@ def create_project(project_name: str, base_directory: DirectoryPath = None) -> N
...
+        base_settings (Dict[str, str]): settings to apply
+    """
+    with open(file_path, "w") as fh:
+        for key, value in base_settings.items():
+            fh.write(f'{key} = "{value}"')
 
-    project_info_location = os.path.join(base_directory, CDEV_FOLDER, CDEV_PROJECT_FILE)
 
-    local_project(project_info_location)
+def _touch_file(file_path: FilePath):
```
Comment (lo9i) [inline src/cdev/commands/initializer.py:258]: Return type hint

**Review by lo9i — CHANGES_REQUESTED** [review]

**Review by cleoberf — APPROVED** [review]

**Pull request merged**